ebu_tt_live/ui/user_input_producer/uip.js

```diff
@@ -0,0 +1,570 @@
+                  var scheduled_time_parsed = scheduled_time_input.match(/(\d\d):(\d\d):(\d\d)/);
+                  if (scheduled_time_parsed == null)
+                     return 0;
+                  scheduled_time += parseInt(scheduled_time_parsed[1]) * 3600000;
+                  scheduled_time += parseInt(scheduled_time_parsed[2]) * 60000;
+                  scheduled_time += parseInt(scheduled_time_parsed[3]) * 1000;
+                  timeout = scheduled_send_media_clock_offset + scheduled_time;
+                  timeout = timeout - Date.now();
+               }
+               return timeout;
+            }
+
+            function submitDocument() {
+               var sending_type = $("input[name=sending-type-radio-input]:checked").val();
+               if (sending_type == "scheduled_send") {
+                  /* If we are using sheduled time to send documents we must actually compute a timeout
+                   * value which is the difference between the scheduled time and the current time.
+                   */
+                  var media_offset = null;
+                  var scheduled_send_clock_type = $("#scheduled-send-clock-selector").val();
+                  if (scheduled_send_clock_type == "media")
+                     media_offset = scheduled_send_media_clock_offset;
+                  var timeout = computeScheduledSendTimeout(media_offset);
+                  var template_data = createTemplateDict();
+                  var rendered_document = nunjucks.render('ebu_tt_live/ui/user_input_producer/template/user_input_producer_template.xml', template_data);
+                  setTimeout(renderSendDocument, timeout, rendered_document);
+                  notifySuccess($("#scheduled-confirmation-span"), "Scheduled...", true);
+                  sequence_numbers[sequence_identifier] += 1;
+               } else {
+                  renderSendDocument();
+                  sequence_numbers[sequence_identifier] += 1;
+               }
+               localStorage.sequence_numbers = JSON.stringify(sequence_numbers);
+            }
+
+            function asyncSubmit() {
+               var template_data = createTemplateDict();
+               var rendered_document = nunjucks.render('ebu_tt_live/ui/user_input_producer/template/user_input_producer_template.xml', template_data);
+               renderSendDocument(rendered_document);
+               sequence_numbers[sequence_identifier] += 1;
+               localStorage.sequence_numbers = JSON.stringify(sequence_numbers);
+            }
+
+            function renderSendDocument(doc=null) {
+               var template_data = createTemplateDict();
+               var rendered_document = null;
+               // If doc in not null then it was already rendered (needed for scheduled times : the document
+               // has to be rendered when the sending is scheduled not at effective sending time.
+               if (doc == null)
```

Comment:
Brackets for style, why not check if undefined rather than setting null at top of function?